src/core/Akka.Remote.Tests/Transport/HeliosTransportDnsResolutionSpec.cs

```diff
@@ -158,6 +166,56 @@ public Property HeliosTransport_Should_Resolve_DNS(EndPoint inbound, EndPoint ou
             }
         }
 
+        [Property]
+        public Property HeliosTransport_Should_Resolve_DNS_with_PublicHostname(IPEndPoint inbound, DnsEndPoint publicInbound,
+            IPEndPoint outbound, DnsEndPoint publicOutbound, bool dnsUseIpv6)
+        {
+            if (dnsUseIpv6 &&
```

Comment:
Filter out invalid combinations as pre-conditions for the spec. Requires the end-user to be consistent about whether they're using IPV4 or IPV6 for DNS resolution.